app/src/main/java/com/infomaniak/swisstransfer/ui/screen/main/transferdetails/TransferDetailsScreen.kt

```diff
@@ -298,7 +299,7 @@ private fun Preview(@PreviewParameter(TransferUiListPreviewParameter::class) tra
                 transferUrl = "",
                 direction = TransferDirection.SENT,
                 navigateBack = null,
-                getTransfer = { transfers.first() },
+                getTransfer = { Success(transfers.first()).transfer },
```

Comment:
Why do we create a `data class` only to get the Transfer that we already have ?

Reply:
How I see things is that the "data class" here is first used to convey the state of the ui. 

Then, when the state is "success", we also want to know what transfer to display in the UI and can use this `Success` data class to pass it down

Reply:
I like to think about it like having an enum except that you can have different signatures for each value of the enum. This way you can have only relevant data inside each value of the "enum".

But don't take my words for it, I'd still be curious about the other's answer

Reply:
Nevermind you didn't ask bout the UiState